cmake/bintools/gnu/target_bintools.cmake

```diff
@@ -62,15 +62,19 @@ set_property(TARGET bintools PROPERTY elfconvert_flag_outfile "")
 # - disassembly : Name of command for disassembly of files
 #                 In this implementation `objdump` is used
 #   disassembly_flag               : -d
-#   disassembly_flag_final         : empty
+#   disassembly_flag_final         : empty or -M no-aliases
 #   disassembly_flag_inline_source : -S
 #   disassembly_flag_all           : -SDz
 #   disassembly_flag_infile        : empty, objdump doesn't take arguments for filenames
 #   disassembly_flag_outfile       : '>', objdump doesn't take arguments for output file, but result is printed to standard out, and is redirected.
 
 set_property(TARGET bintools PROPERTY disassembly_command ${CMAKE_OBJDUMP})
 set_property(TARGET bintools PROPERTY disassembly_flag -d)
+if(CONFIG_OUTPUT_DISASSEMBLY_NO_ALIASES)
+set_property(TARGET bintools PROPERTY disassembly_flag_final "-M no-aliases")
+else()
 set_property(TARGET bintools PROPERTY disassembly_flag_final "")
+endif()
```

Comment:
This is not following the existing design pattern.

A correct implementation would introduce a new bintools "property" called `disassembly_flag_no_aliases`, which is set by each bintool and consumed by the root `CMakeLists.txt` in:

https://github.com/zephyrproject-rtos/zephyr/blob/4cae4661a262215ee288041371b4daa84dfabc9e/CMakeLists.txt#L1901-L1920